internal/k8s/reconciler/gateway.go

```diff
@@ -464,8 +464,19 @@ func (g *K8sGateway) ensureDeploymentExists(ctx context.Context) error {
 
 	deployment := g.deploymentBuilder.Build()
 	mutated := deployment.DeepCopy()
+
+	currentDeployment, err := g.client.GetDeployment(ctx, types.NamespacedName{Namespace: deployment.Namespace, Name: deployment.Name})
+	if err != nil {
+		return err
+	}
+
+	var currentReplicas *int32
+	if currentDeployment != nil {
+		currentReplicas = currentDeployment.Spec.Replicas
+	}
+
```

Comment:
I think a comment explaining why we're fetching the deployment and replica count would be useful